In [84]:
#import library
import os
import csv
from pathlib import Path
import pandas as pd
import numpy as np


In [85]:
# Set path to collect data 
data = os.path.exists('2019 Winter Data Science Intern Challenge Data Set.xlsx')
# Make a reference to the data
path = '2019 Winter Data Science Intern Challenge Data Set.xlsx'

In [86]:
# Import the data set as a DataFrame
data_df = pd.read_excel(path) 
data_df.head()

order_id  shop_id  user_id  order_amount  total_items payment_method  \
0         1       53      746           224            2           cash   
1         2       92      925            90            1           cash   
2         3       44      861           144            1           cash   
3         4       18      935           156            1    credit_card   
4         5       18      883           156            1    credit_card   

               created_at  
0 2017-03-13 12:36:56.190  
1 2017-03-03 17:38:51.999  
2 2017-03-14 04:23:55.595  
3 2017-03-26 12:43:36.649  
4 2017-03-01 04:35:10.773

In [43]:
#total revenue
tot_rev=data_df['order_amount'].sum()
#total items
tot_items=data_df['total_items'].sum()
#AOV
AOV=tot_rev/tot_items
print("Total revenue generated:",tot_rev)
print("Total items sold:",tot_items)
print("Average order value:",AOV)

Total revenue generated: 15725640
Total items sold: 43936
Average order value: 357.92152221412965


# Answer 1


In [88]:
#total revenue
tot_rev=data_df['order_amount'].sum()
#total items error
tot_items=data_df['total_items'].count()
#AOV
AOV=tot_rev/tot_items
print("Total revenue generated:",tot_rev)
print("Total items sold:",tot_items)
print("Average order value:",AOV)

Total revenue generated: 15725640
Total items sold: 5000
Average order value: 3145.128


Therefore the error in calculating total_items results in error. Instead of summing the value of items, count was used.

# Answer 2


To correctly calculate AOV (average order value) we need to divide the total revenue in the time period with the total number of items sold. That is, AOV = total_revenue/total_items

# Answer 3


The correct value is 357.92